src/ucp/proto/proto_am.inl

```diff
@@ -28,12 +28,19 @@ ucp_do_am_bcopy_single(uct_pending_req_t *self, uint8_t am_id,
 {
     ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
     ucp_ep_t *ep       = req->send.ep;
+    size_t offset      = req->send.state.dt.offset;
     ssize_t packed_len;
 
     req->send.lane = ucp_ep_get_am_lane(ep);
     packed_len     = uct_ep_am_bcopy(ep->uct_eps[req->send.lane], am_id, pack_cb,
                                      req, 0);
-    if (packed_len < 0) {
+    ucs_assertv((packed_len < 0) ||
+                (packed_len <= ucp_ep_get_max_bcopy(ep, req->send.lane)),
```

Comment:
may be condition 
```ucs_assertv(packed_len <= ucp_ep_get_max_bcopy(ep, req->send.lane),...``` is enough?

Reply:
yes, it would be enough
but compilers won't be happy with that since different sugndness for `packed_len` (signed) and the result of `ucp_ep_get_max_bcopy()` (unsigned)

Reply:
then ```ucs_assertv(packed_len <= (ssize_t)ucp_ep_get_max_bcopy(ep, req->send.lane),...```

Reply:
yes, but imaging that max_bcopy can be larger than SSIZE_MAX, I think it can be broken